objects/src/transaction/utils.rs

```diff
@@ -128,3 +127,15 @@ pub fn generate_created_notes_commitment(notes: &[Note]) -> Digest {
 
     Hasher::hash_elements(&elements)
 }
+
+/// Returns the created notes commitment.
+/// This is a sequential hash of all (hash, metadata) pairs for the notes created in the transaction.
+pub fn generate_created_notes_stub_commitment(notes: &[NoteStub]) -> Digest {
+    let mut elements: Vec<Felt> = Vec::with_capacity(notes.len() * 8);
+    for note in notes.iter() {
+        elements.extend_from_slice(note.hash().as_elements());
+        elements.extend_from_slice(&Word::from(note.metadata()));
+    }
+
+    Hasher::hash_elements(&elements)
+}
```

Comment:
Seems like this function is used only in the `CreatedNotes` constructor - is there a reason not to put it into the same module?